--- a/lib_internal/util-cleanupmount.c
+++ b/lib_internal/util-cleanupmount.c
@@ -25,7 +25,6 @@
 #ifdef HAVE_LINUX_TYPES_H
 #  include <linux/types.h>
 #endif
-#include <linux/fs.h>
 
 #ifndef MS_REC
 #define MS_REC		0x4000
--- a/src/secure-mount.c
+++ b/src/secure-mount.c
@@ -49,7 +49,7 @@
 #ifdef HAVE_LINUX_TYPES_H
 #  include <linux/types.h>
 #endif
-#include <linux/fs.h>
+#include <limits.h>
 #include <assert.h>
 #include <ctype.h>
 #include <sys/wait.h>
